core/src/main/java/hudson/model/Job.java

```diff
@@ -1143,6 +1148,22 @@ private HealthReport getBuildStabilityHealthReport() {
         int failCount = 0;
         int totalCount = 0;
         RunT i = getLastBuild();
+        RunT u = getLastFailedBuild();
+        if (i != null && u == null) {
+                // no failures, like ever
```

Comment:
All this logic is fragile in the case of concurrent builds, unfortunately. BTW, it's not a new defect